test/staged.jl

```diff
@@ -174,7 +174,7 @@ let gf_err, tsk = @async nothing # create a Task for yield to try to run
     end
     @test_throws ErrorException gf_err()
     @test_throws ErrorException gf_err()
-    @test gf_err_ref[] == 3
+    @test gf_err_ref[] == 4
```

Comment:
why does the number of times this attempted to and failed to run the generated function change then?